Makefile

```diff
@@ -15,16 +15,17 @@ endef
 
 # Generate all implementations
 .PHONY: gen-all
-gen-all: gen-cpp gen-csharp gen-go gen-java gen-objc gen-openapi gen-php gen-python gen-ruby
+gen-all: gen-cpp gen-csharp gen-go gen-java gen-objc gen-openapi gen-php gen-python gen-ruby gen-kotlin
 
-OTEL_DOCKER_PROTOBUF ?= otel/build-protobuf:0.4.0
+OTEL_DOCKER_PROTOBUF ?= otel/build-protobuf:0.7.0
```

Comment:
As per https://github.com/open-telemetry/build-tools/releases this should be a no-op change anyway but just to double-check:
@bogdandrutu @tigrannajaryan are we fine updating this? The next version (0.8.0) will also bump the alpine image and Go version (1.15->1.17).

Reply:
The build failed again so I looked into the image an I see that the version used is quite old: https://github.com/open-telemetry/build-tools/blob/cccfc34393ba3b71ca2956060b7d351b38564945/protobuf/Dockerfile#L5
This version needs to be updated to v3.17.0 at least for this to be able to create the native Kotlin classes.
I can also update the Dockerfile though it might have some implications

Reply:
It is ok to upgrade the image.